### ISSUE TITLE
Any is excessively used as outer context for inference

### ISSUE THREAD

**Issue by ilevkivskyi** [python/mypy#8760]
Consider this example:
```python
from typing import Any
  
def foo(**kwargs: Any) -> None: ...

foo(bar=reveal_type([1])) # Revealed type is List[Any]
```
Such behavior causes many false negatives for plugins that provide precise type checking for various ORMs, this includes SQLAlchemy plugin, our internal plugins, and likely Django plugin.

My _guess_ this was added so that an empty list in `Any` context is inferred as `List[Any]` instead of `List[<nothing>]`, but I think this is counterproductive. If empty collections cause issues for plugins they can easily work around using private APIs to re-infer argument type in generated precise context (for example using `has_uninhabited_component()` and `ctx.api.accept()`). Or maybe we can provide public API for inference.

**Comment by ilevkivskyi** [comment]
Actually there are two possible workaround (at least for some plugins):
* Use `**kwargs: object` annotation in relevant stubs.
* Switch to `get_method_signature_hook()` instead of `get_method_hook()`.

So probably this is not high priority.
